Logic.py

```diff
@@ -71,3 +72,706 @@ def has_qwark_statuette(state: CollectionState, player: int) -> bool:
+
+    return can_gravity(state, player)
+
+
+def snivelak_rescue_angelak_rule(state: CollectionState, player: int) -> bool:
+    options = get_options(state, player)
+
+    if options.allow_first_person_mode_glitch_locations >+ FIRST_PERSON_EASY:
+        return can_swingshot(state, player)
+
+    return (can_swingshot(state, player)
+            and can_grind(state, player)
+            and can_gravity(state, player)
+            and can_dynamo(state, player))
+
+
+def snivelak_dynamo_pb_rule(state: CollectionState, player: int) -> bool:
+    options = get_options(state, player)
+
+    if options.allow_first_person_mode_glitch_locations >= FIRST_PERSON_EASY:
+        return can_swingshot(state, player)
+
+    return (can_swingshot(state, player)
+            and can_grind(state, player)
+            and can_gravity(state, player)
+            and can_dynamo(state, player)
+            and can_heli(state, player))
+
+
+def snivelak_swingshot_tower_nt_rule(state: CollectionState, player: int) -> bool:
+    options = get_options(state, player)
+
+    if options.allow_first_person_mode_glitch_locations >= FIRST_PERSON_EASY:
+        return can_swingshot(state, player)
+
+    return (can_swingshot(state, player)
+            and can_heli(state, player))
+
+
+def smolg_balloon_transmission_rule(state: CollectionState, player: int) -> bool:
+    options = get_options(state, player)
+
+    if options.allow_first_person_mode_glitch_locations >= FIRST_PERSON_EASY:
+        return can_electrolyze(state, player)
+
+    return (can_improved_jump(state, player)
+            and can_dynamo(state, player)
+            and can_electrolyze(state, player))
+
+
+def smolg_distribution_facility_end_rule(state: CollectionState, player: int) -> bool:
+    options = get_options(state, player)
+
+    if options.allow_first_person_mode_glitch_locations >= FIRST_PERSON_EASY:
+        return can_electrolyze(state, player)
+
+    return (can_improved_jump(state, player)
+            and can_dynamo(state, player)
+            and can_electrolyze(state, player)
+            and can_grind(state, player)
+            and can_infiltrate(state, player))
+
+
+def smolg_mutant_crab_rule(state: CollectionState, player: int) -> bool:
+    options = get_options(state, player)
+
+    if options.allow_first_person_mode_glitch_locations >= FIRST_PERSON_EASY:
+        if not can_levitate(state, player):
+            return False
+        return (can_swingshot(state, player)
+                or can_electrolyze(state, player))
+
+    return (can_swingshot(state, player)
+            and can_levitate(state, player))
+
+
+def smolg_floating_platform_pb_rule(state: CollectionState, player: int) -> bool:
+    options = get_options(state, player)
+
+    if options.allow_first_person_mode_glitch_locations >= FIRST_PERSON_EASY:
+        if not can_levitate(state, player):
+            return False
+        return (can_swingshot(state, player)
+                or can_electrolyze(state, player))
```

Comment:
Wouldn't something like this be equivalent?
```suggestion
    if options.allow_first_person_mode_glitch_locations >= FIRST_PERSON_EASY:
        return (can_levitate(state, player)
                and (
                can_swingshot(state, player) 
                or can_electrolyze(state, player)))
```

Reply:
They are equivalent, indeed.
I just wanted to emphasize the need for the Levitator.
I can change it if needed.